index.php

```diff
@@ -44,13 +43,12 @@
     array_unshift($templates, 'all-posts.twig');
 
     if (!empty(planet4_get_option('new_ia'))) {
-        $view = ListingPageGridView::is_active() ? 'grid' : 'list';
-
-        $query_template = file_get_contents(get_template_directory() . "/parts/query-$view.html");
-
-        $content = do_blocks($query_template);
-
-        $context['query_loop'] = $content;
+        $grid_template = file_get_contents(get_template_directory() . "/parts/query-grid.html");
```

Comment:
What do you think if we create a static PHP class to get all these code which is being repeated also in other listing pages?

Reply:
I thought about it, I agree it's not ideal having duplicated code like this... But I think it should be handled in another PR, we could create a follow-up maintenance ticket

Reply:
Created a [ticket](https://jira.greenpeace.org/browse/PLANET-7365) for this